src/core/server/elasticsearch/cluster_client.ts

```diff
@@ -66,7 +70,16 @@ async function callAPI(
 
   const apiContext = clientPath.length === 1 ? client : get(client, clientPath.slice(0, -1));
   try {
-    return await api.call(apiContext, clientParams);
+    return await new Promise((resolve, reject) => {
+      // We have to use a callback here instead of using promises directly because it's the only way
+      // to get access to the abort method on the request
+      const request = api.call(apiContext, clientParams, (err: any, res: any) =>
+        err ? reject(err) : resolve(res)
+      );
+      if (options.signal) {
+        options.signal.onabort = () => request.abort();
```

Comment:
Not a requirement, but if you want to achieve this without mutating `signal` and therefore `options`, you can use:

```js
options.signal.addEventListener('abort', () => request.abort());
```